Sources/SwiftDocC/Infrastructure/Link Resolution/LinkResolver.swift

```diff
@@ -0,0 +1,183 @@
+    private let fallbackResolver = FallbackResolverBasedLinkResolver()
+    /// A map of link resolvers for external, already build archives
+    var externalResolvers: [String: ExternalPathHierarchyResolver] = [:]
+    
+    /// Create link resolvers for all documentation archive dependencies.
+    func loadExternalResolvers() throws {
+        let resolvers = try dependencyArchives.compactMap {
+            try ExternalPathHierarchyResolver(dependencyArchive: $0)
+        }
+        for resolver in resolvers {
+            for moduleName in resolver.pathHierarchy.modules.keys {
+                self.externalResolvers[moduleName] = resolver
+            }
+        }
+    }
+    
+    // ???: Should this be aliased the other way around?
+    typealias ExternalEntity = ExternalPathHierarchyResolver.ExternalEntity
+    
+    /// Attempts to resolve an unresolved reference.
+    ///
+    /// - Parameters:
+    ///   - unresolvedReference: The unresolved reference to resolve.
+    ///   - parent: The parent reference to resolve the unresolved reference relative to.
+    ///   - isCurrentlyResolvingSymbolLink: Whether or not the documentation link is a symbol link.
+    ///   - context: The documentation context to resolve the link in.
+    /// - Returns: The result of resolving the reference.
+    func resolve(_ unresolvedReference: UnresolvedTopicReference, in parent: ResolvedTopicReference, fromSymbolLink isCurrentlyResolvingSymbolLink: Bool, context: DocumentationContext) -> TopicReferenceResolutionResult {
+        // Check if the unresolved reference is external
+        if let bundleID = unresolvedReference.bundleIdentifier,
+           !context.registeredBundles.contains(where: { bundle in
+               bundle.identifier == bundleID || urlReadablePath(bundle.displayName) == bundleID
+           }) {
+            
+            if context.externalReferenceResolvers[bundleID] != nil,
+               let resolvedExternalReference = context.externallyResolvedLinks[unresolvedReference.topicURL] {
+                // Return the successful or failed externally resolved reference.
+                return resolvedExternalReference
+            } else if !context.registeredBundles.contains(where: { $0.identifier == bundleID }) {
+                return .failure(unresolvedReference, TopicReferenceResolutionErrorInfo("No external resolver registered for \(bundleID.singleQuoted)."))
+            }
+        }
+        
+        do {
+            return try localResolver.resolve(unresolvedReference, in: parent, fromSymbolLink: isCurrentlyResolvingSymbolLink, context: context)
+        } catch let error as PathHierarchy.Error {
+            // Check if there's a known external resolver for this module.
+            if case .moduleNotFound(let remaining, _) = error, let resolver = externalResolvers[remaining.first!.full] {
+                if let previousResult = context.externallyResolvedLinks[unresolvedReference.topicURL] {
```

Comment:
If we keep the external links prefixed with a "/" then we could do that. Otherwise there's a change that the link will resolve to a local symbol in a different parent context.

Reply:
Ah got it. Sounds good then 👍🏻

Reply:
I changed this to check this before trying the local resolver